### DISCUSSION TITLE
An internal server error occurred

### DISCUSSION THREAD

**Discussion by GabrielLiade** [strapi/strapi#7651] [Legacy Topics]
Hi everyone, I have" An internal server error occurred " when I'm trying to connect to my Strapi back deployed on Heroku (it works well when is local) any ideas of where the error may come from

I don't know, what more information I can give you but feel free to ask ;)

thank you in advance

**Comment by derrickmehaffy** [comment]
Can you please provide any relevant logs from heroku? It will be fairly difficult to troubleshoot with just a 500 error 😉

**Reply by GabrielLiade** [reply]
yes I would like but how can I see it on Heroku 
thank a lot for your answer !

**Reply by derrickmehaffy** [reply]
https://devcenter.heroku.com/articles/logging

**Reply by GabrielLiade** [reply]
when I re-clone my Strapi back localy on my computer and the internal error came back 

this is what the console tell me : 

error Error: secretOrPrivateKey must have a value
    at Object.module.exports [as sign] (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/jsonwebtoken/sign.js:107:20)
    at Object.createJwtToken (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/strapi-admin/services/token.js:32:14)
    at /Users/gabrieleliade/Desktop/wtc-back-api/node_modules/strapi-admin/controllers/authentication.js:35:46
    at dispatch (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/koa-compose/index.js:42:32)
    at /Users/gabrieleliade/Desktop/wtc-back-api/node_modules/strapi-admin/controllers/authentication.js:27:16
    at callback (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/koa-passport/lib/framework/koa.js:93:25)
    at Strategy.strategy.success (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/passport/lib/middleware/authenticate.js:219:18)
    at verified (/Users/gabrieleliade/Desktop/wtc-back-api/node_modules/passport-local/lib/strategy.js:83:10)
    at /Users/gabrieleliade/Desktop/wtc-back-api/node_modules/strapi-admin/middlewares/auth/index.js:16:43
[2020-08-31T15:17:51.295Z] debug POST /admin/login (136 ms) 500

**Reply by GabrielLiade** [reply]
I hope it's enough to clear the bug =(
thank a lot to take time with me ;)

**Reply by derrickmehaffy** [reply]
You are missing an `ADMIN_JWT_SECRET` from the `./config/server.js` this is something you need to generate and set as an environment variable or locally you can use the `.env` file.

From our migration provides some ways to generate one: https://strapi.io/documentation/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.html#_2-define-the-admin-jwt-token

**Reply by GabrielLiade** [reply]
It's what I was thinking about
The back start to bug when I did the migration to the 3.1 of Strapi, but my website is live so I can't keep it with the bug aha. So I came back to the 3.0
I will try your advices (on a preprod this time :)) and I keep you posted 
thank you so much !

**Reply by derrickmehaffy** [reply]
Yeah you need to follow the full migration guide that I linked above, any time you are updating a major version such as `3.x` to `3.1` you should check for a migration guide.

**Reply by GabrielLiade** [reply]
I have to put the JWT secret in the config var in Heroku ?

**Reply by derrickmehaffy** [reply]
Yes, whatever the variable is named in your `server.js` by default it's `ADMIN_JWT_SECRET`
